src/test/scala/edu/berkeley/ce/rockslicing/InputProcessingSpec.scala

```diff
@@ -5,169 +5,267 @@ import org.scalatest.FunSuite
 import scala.io.Source
 
 class InputProcessingSpec extends FunSuite {
+
+  def compareJointSets(jointSet1: JointSet, jointSet2: JointSet): Boolean = {
```

Comment:
I think, if you make the changes I've suggested above, you can compare the `JointSet`s directly with `==`. Case classes give you some handy methods for free.